server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -304,14 +304,7 @@ public DiscoveryNode resolveNode(String node) {
      *   or a generic node attribute name in which case value will be treated as a wildcard and matched against the node attribute values.
      */
     public String[] resolveNodes(String... nodes) {
-        if (isAllNodes(nodes)) {
-            int index = 0;
-            nodes = new String[this.nodes.size()];
-            for (DiscoveryNode node : this) {
-                nodes[index++] = node.getId();
-            }
-            return nodes;
-        } else {
+        {
```

Comment:
This is deliberate so as to not change the indentation in this pull request so that the change is easier to review. A separate change can go in that removes the indentation and is a formatting-only change.